client/src/elm/Pages/NCD/RecurrentActivity/Update.elm

```diff
@@ -0,0 +1,705 @@
+                    getMeasurementValueFunc saved
+
+                extraMsgs =
+                    generateLabResultsMsgs nextTask
+
+                appMsgs =
+                    toCreatinineResultsValueWithDefault measurement model.labResultsData.creatinineTestForm
+                        |> Maybe.map
+                            (Backend.NCDEncounter.Model.SaveCreatinineTest personId measurementId
+                                >> Backend.Model.MsgNCDEncounter id
+                                >> App.Model.MsgIndexedDb
+                                >> List.singleton
+                            )
+                        |> Maybe.withDefault []
+            in
+            ( model
+            , Cmd.none
+            , appMsgs
+            )
+                |> sequenceExtra (update language currentDate id db) extraMsgs
+
+        SetAltResult value ->
+            let
+                form =
+                    model.labResultsData.liverFunctionTestForm
+
+                updatedForm =
+                    { form | altResult = String.toFloat value }
+
+                updatedData =
+                    model.labResultsData
+                        |> (\data -> { data | liverFunctionTestForm = updatedForm })
+            in
+            ( { model | labResultsData = updatedData }
+            , Cmd.none
+            , []
+            )
+
+        SetAstResult value ->
+            let
+                form =
+                    model.labResultsData.liverFunctionTestForm
+
+                updatedForm =
+                    { form | astResult = String.toFloat value }
+
+                updatedData =
+                    model.labResultsData
+                        |> (\data -> { data | liverFunctionTestForm = updatedForm })
+            in
+            ( { model | labResultsData = updatedData }
+            , Cmd.none
+            , []
+            )
+
+        SaveLiverFunctionResult personId saved nextTask ->
+            let
+                measurementId =
+                    Maybe.map Tuple.first saved
+
+                measurement =
+                    getMeasurementValueFunc saved
+
+                extraMsgs =
+                    generateLabResultsMsgs nextTask
+
+                appMsgs =
+                    toLiverFunctionResultsValueWithDefault measurement model.labResultsData.liverFunctionTestForm
+                        |> Maybe.map
+                            (Backend.NCDEncounter.Model.SaveLiverFunctionTest personId measurementId
+                                >> Backend.Model.MsgNCDEncounter id
+                                >> App.Model.MsgIndexedDb
+                                >> List.singleton
+                            )
+                        |> Maybe.withDefault []
+            in
+            ( model
+            , Cmd.none
+            , appMsgs
+            )
+                |> sequenceExtra (update language currentDate id db) extraMsgs
+
+        -- SetActiveNextStepsTask task ->
```

Comment:
todo or eliminate ?

Reply:
Next steps related development is done at https://github.com/TIP-Global-Health/eheza-app/pull/456 and https://github.com/TIP-Global-Health/eheza-app/pull/459, All commented out code is handled there.